subprojects/packagefiles/openssl/generator.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+set -e
+
+cd "$(dirname "${BASH_SOURCE[0]}")"
```

Comment:
Hum, this is wrong. `run_command()` is run from an unspecified directory, and you are supposed to use `MESON_SOURCE_ROOT`, `MESON_BUILD_ROOT` and `MESON_SUBDIR` to position yourself in the right place.

Reply:
This script is used in 2 cases:
1) When called manually and directly, this is useful when wrap release archive needs to be created with generated files included in it
2) Automatically from `meson.build` for convenience such that generator script is not needed to be called explicitly before `meson setup`

In first case those environment variables are irrelevant, in second case it happens to generate files in `subprojects/openssl-1.1.1k`, which I agree isn't nice, but at the same time I'm not sure what to do there properly, the end result is that we need those generated files to proceed, ideally in `subprojects/openssl-1.1.1k` such that we don't have more special cases just to make working with `wrapdb` more convenient.

Reply:
It's fine to generate those files into the source dir, that's probably the correct place for them in this case. All I am saying is that the script should not embed assumptions about where it is being called from when it is called by Meson.

So it should check if `MESON_SOURCE_ROOT` is set, and use `MESON_SUBDIR` to find the `subprojects/openssl-1.1.1k` directory and put things in there. If `MESON_SOURCE_ROOT` is not set, then it has been called manually and it can do what it is doing right now.

Reply:
Oh, I misunderstood how `BASH_SOURCE` works, sorry. You're always generating the sources into the same directory as `generator.sh` that's fine I guess and will always be correct.